src/ducks/Signals/chart/VisualBacktestChart.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react'
+import {
+  ResponsiveContainer,
+  ComposedChart,
+  XAxis,
+  YAxis,
+  Tooltip,
+  ReferenceDot,
+  ReferenceLine
+} from 'recharts'
+import cx from 'classnames'
+import { generateMetricsMarkup } from './../../SANCharts/utils'
+import { formatNumber } from './../../../utils/formatting'
+import { getDateFormats, getTimeFormats } from '../../../utils/dates'
+import chartStyles from './../../SANCharts/Chart.module.scss'
+import sharedStyles from './../../SANCharts/ChartPage.module.scss'
+import styles from './SignalPreview.module.scss'
+
+export const getDataKeys = (signal = {}) => {
+  if (signal.active_addresses) {
+    return { metric: 'dailyActiveAddresses', signal: 'active_addresses' }
+  }
+  if (signal.price) return { metric: 'priceUsd', signal: 'price' }
+  return { metric: 'priceUsd' }
+}
+
+const mapWithTimeseries = items =>
+  items.map(item => ({ ...item, datetime: +new Date(item.datetime) }))
```

Comment:
We do it in so many places. Maybe we can do it in GetTimeSeries during fetching data. But it's just thought.

Reply:
l'll investigate